src/layout.jl

```diff
@@ -635,36 +630,67 @@ PlaceHolder(ren::AbstractRenderable) = PlaceHolder(ren.measure.w, ren.measure.h)
 #                                     GRID                                     #
 # ---------------------------------------------------------------------------- #
 
+"""
+    calc_nrows_ncols(n, aspect=(16, 9))
+
+Computes the number of rows and columns to fit a given number `n` of subplots in a figure with aspect `aspect`.
+Adapted from: stackoverflow.com/a/43366784
+"""
+function calc_nrows_ncols(n, aspect=(16, 9))
+    w, h = aspect
+    factor = √(n / (w * h * 1.0))
+    cols = floor(Int, w * factor)
+    rows = floor(Int, h * factor)
+    rowFirst = w < h
+    while rows * cols < n
+        if rowFirst
+            rows += 1
+        else
+            cols += 1
+        end
+        rowFirst = !rowFirst
+    end
+    return rows, cols
+end
+
 """
     grid(rens::AbstractVector{<:AbstractRenderable}, layout::Union{Nothing,Tuple} = nothing, placeholder = nothing)
 
-Construct a square grid from a `AbsractVector` of `AbstractRenderable`
+Construct a square grid from a `AbsractVector` of `AbstractRenderable`.
 """
 function grid(
     rens::AbstractVector{<:AbstractRenderable};
     layout::Union{Nothing,Tuple} = nothing, 
-    placeholder = nothing
+    placeholder = nothing,
+    aspect = (16, 9),
+    hspacer = "",
+    vspacer = hspacer,
 )
     num = length(rens)
-    if layout === nothing
-        m = ceil(Int, √(num))
+    m, n = layout === nothing ? calc_nrows_ncols(num, aspect) : layout
```

Comment:
Trying to learn something here - your Julia is definitely much better than mine!
What's the advantage of using `layout === nothing` over `isnothing(layout)`?

Reply:
AFAIK , in the old days, `=== nothing` was better for the compiler. Also, using this syntax was to retain compatibility with the older julia versions, before `isnothing` was introduced. So nowadays, it is only a matter of personal taste and code style: I see now that your are already using `isnothing`, I will change to that.

Reply:
I see, thanks!

Reply:
> What's the advantage of using layout === nothing over isnothing(layout)?

They're exactly the same thing if you check `@less isnothing(1)`. One small difference is that `isnothing` is introduced in Julia 1.1 so if you're going to maintain 1.0 compatibility, you should use `===`. But since Term.jl requires Julia at least 1.6, just using `isnothing` should be fine -- people introduce and recommend `isnothing` because it's not immediately obvious to new Julia users that `x === nothing` is faster than `x == nothing`.

In Julia >= 1.7, there's also an equivalent version `m, n = @something layout calc_nrows_ncols(num, aspect)` that returns the first non-nothing value.